helpers/embeds.js

```diff
@@ -0,0 +1,78 @@
+import { findComposedAncestor } from '../helpers/dom.js';
+import { html } from 'lit';
+import { tryGetPluginByKey } from '@brightspace-ui/core/helpers/plugins.js';
+
+class RenderNode {
+	#childNodes;
+	#node;
+
+	constructor(node) {
+		this.#node = node;
+		this.#childNodes = [];
+	}
+
+	addChildNode(childNode) {
+		this.#childNodes.push(childNode);
+	}
+
+	async renderNode(options) {
+		for await (const childNode of this.#childNodes) {
+			await childNode.renderNode(options);
+		}
+
+		if (this.#node.id === '3') {
+			console.log('wee');
```

Comment:
I was just using this for debugging. Probably expect commits to show up with debugging code still in it while this is a PoC. I'm not exactly being careful about making sure my code is up to snuff. 😆